utils/generator/README.md

```diff
@@ -0,0 +1,2245 @@
+
+## The License Header
+
+All files should start from the comment with the license information.
+
+```java
+/*
+ * Copyright (c) 2017 - [year], SmartDeviceLink Consortium, Inc.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the SmartDeviceLink Consortium Inc. nor the names of
+ * its contributors may be used to endorse or promote products derived
+ * from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+```
+Where `[year]` in the copyright line is the current year.
+
+## `<enum>`
+
+The name of the class is the value from the `"name"` attribute of `<enum>`.
+
+The class should have the next JavaDoc comment:
+```java
+/**
+ * [description]
+ *
+ * @deprecated
+ * @since SmartDeviceLink [since_version]
+ * @see [see_reference]
+ */
+```
+Where:
+* `[description]` is `<description>` of the current `<enum>`, if exists.
+* `@deprecated` indicates the deprecation state if the `"deprecated"` attribute exists and is "true".
+* `@since` should be present, if the `"since"` attribute exists, and `[since_version]` is the `Major.Minor.Patch` formatted value of this attribute.
+* `@see` shows the custom reference in `[see_reference]`.
+
+The class should have the `@Deprecated` decorator if the `"deprecated"` attribute of the `<enum>` exists and is "true".
+
+The set of `<element>` should be mapped to the set of Enum constants. Based on the `<element>` attributes, constants could be with or without fields.
+
+The following list are general rules for constant names and its fields:
+1. The `"name"` attribute of `<element>` is the default name of the constant.
+    * if the name starts from digit, the leading `_` (underscore) symbol should be added.
```

Comment:
I think that case doesn't exist anymore in the code. Also, I believe, the case where we replace "-" with "_" is not mentioned in the readme.

Reply:
No, this exists because there is Enums which will fail without it, e.g. BitsPerSample
element name="8_BIT" should be converted to _8_BIT

Reply:
I meant the other point:
> if the name starts with _ (underscore) symbol and the next character is a letter of alphabet, the leading _ (underscore) symbol should be removed.

do we still have this case?

Reply:
Also, I am not sure why, but the CI doesn't pass on this PR.

Reply:
Thanks, Bilal, updated on both questions, waiting for current build to see the problem.

Reply:
Awesome. It passes now. Thanks.

Reply:
As I see from history it was failing either on android artifacts download with 504 error or git checkout which is far from RPC generator changes.

Reply:
Yes, that is strange. it fails again now and keeps failing when I restart it.
Failing on downloading the Android artifacts is normal and happens rarely sometimes but simply restarting the action fixes it. However, I have never seen it failing on checkout before. I am guessing it is something has to do with checking out out the submodule that was added here:
https://github.com/smartdevicelink/sdl_java_suite/blob/07ad3eee3682e0ce9306b523821f0d3e85dbcc95/.github/workflows/android.yml#L13

Maybe try to remove that line (just for testing) and see if it passes the checkout stage. There are other alternatives available online to checkout submodules if that is what is causing the issue.

Reply:
Agree, removed my android.yml changes, let's see if build will be stable without checking out submodules

Reply:
It is still failing on the checkout stage which is strange. I am still guessing it is something has to do with submodules addition because it passes on all other PRs. Can you please try to remove the `.gitmodules` (just temporarily) to see if that makes it pass?

Reply:
It could be a bug in the `checkout` action. I have made PR #1305  to update the actions `android.yml` to the latest version. It is merged into `develop` now. Can you merge the latest develop into your branch to see if that fixes the issue?

Reply:
I was able to fix this with checkout@master action. Details is in last comments of https://github.com/actions/checkout/issues/81
In short, checkout@v2 did not work with submodules at all now, but updated submodule processing was already comitted to master of checkout action. It will be moved soon to v2 tag

Reply:
Awesome. Thanks for doing that research. We can keep the master tag for now and change to v2 later when the fix is merged.

Reply:
@ksologubov can we try v2 tag again? it seems to be fixed now.

Reply:
@bilal-alsharifi  set to v2, works good

Reply:
Awesome. Thanks